terraform/aws/security/aws-cloudtrail-encrypted-with-cmk.yaml

```diff
@@ -0,0 +1,26 @@
+rules:
+- id: aws-cloudtrail-encrypted-with-cmk
+  patterns:
+  - pattern: |
+      resource "aws_cloudtrail" $ANYTHING {
+        ...
+      }
+  - pattern-not-inside: |
+      resource "aws_cloudtrail" $ANYTHING {
+        ...
+        kms_key_id = ...
+        ...
+      }
+  - pattern-regex: (^aws_kms_key\.(.*))

```

Comment:
It looks like the intent of this regex is to filter out cases where `kms_key_id` is set to something with `aws_kms_key` as a prefix. Is that right?

In that case you'll have to group the `pattern-not-inside` and regex under a parent `patterns` clause so that they go together. In addition, this might be easier to do by capturing the right hand side of `kms_key_id` in a metavariable and using `metavariable-pattern` to test it.

Reply:
That is right. Although, in a second thought, that might not be needed at all and can be scrapped altogether. What do you think?